yb-voyager/src/queryissue/queryissue.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queryissue
+
+import (
+	"fmt"
+
+	pg_query "github.com/pganalyze/pg_query_go/v5"
+	"github.com/samber/lo"
+	log "github.com/sirupsen/logrus"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/issue"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/queryparser"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+type ParserIssueDetector struct {
+	// TODO: Add fields here
+	// e.g. store composite types, etc. for future processing.
+}
+
+func NewParserIssueDetector() *ParserIssueDetector {
+	return &ParserIssueDetector{}
+}
+
+func (p *ParserIssueDetector) GetIssues(query string) ([]issue.IssueInstance, error) {
```

Comment:
Maybe this is for later when we try to add other issue-type reporting here like DDL, PLPGSQL..: but just bringing up here in the context: do we need a parameter in this function to return the Issues of specific type only, otherwise the upstream caller needs to filter the issues.

Reply:
Interesting. What are cases where you might want to filter? 
The way I was thinking about it was: You just pass a query and get back all the Issues in that query. Why would we want to filter by type?